addons/okr-daily-script.kp/knowledge.md

```diff
@@ -176,33 +176,46 @@ ms_install_days = ms\
 # combine data
 current = count_by_client_day\
   .join(ms_install_days, on='client_id', how='left')\
-  .join(ms_has_theme, on='client_id', how='left')\
+  .join(ms_has_theme, on=['client_id', 'normalized_channel'], how='left')\
   .drop('submission_date_s3')
 
 
 # add total number of distinct clients per day
 # and total number of ditscint clients with a custom theme
-n_clients = ms.select('client_id').distinct().count()
+n_clients = ms.select(['client_id', 'normalized_channel']).distinct()\
+               .groupby('normalized_channel').count()\
+               .withColumnRenamed('count', 'n_clients')
+
 n_custom_themes = ms_has_theme\
   .filter(ms_has_theme.has_custom_theme >= 0)\
-  .select('client_id').distinct().count()
-current = current\
-  .withColumn('n_custom_theme_clients', fun.lit(n_custom_themes))\
-  .withColumn('n_clients', fun.lit(n_clients))
+  .select(['client_id', 'normalized_channel']).distinct()\
+  .groupby('normalized_channel').count()\
+  .withColumnRenamed('count', 'n_custom_theme_clients')
 
+current = current.join(n_custom_themes, on='normalized_channel')\
+                 .join(n_clients, on='normalized_channel')
+    
+current = current.withColumn('n_clients', current.n_clients.cast(st.IntegerType()))\
+                 .withColumn('n_custom_theme_clients', current.n_custom_theme_clients.cast(st.IntegerType()))
+    
 # repartition data
 current = optimize_repartition(current, record_size=38)
```

Comment:
I estimated this record size again (total bytes on s3 / # rows for a day) for two different days and got 39 both times. Using the previous record size the partition sizes vary from around (roughly) 250 MiB (262 MB) to 275 MiB (288 MB) -- which seemed OK to me. I think updating to 39 should be fine as it won't drastically change things and it will be consistent with the instructions on how to calculate record size in the `optimize_repartition` docstring.